modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java

```diff
@@ -3308,6 +3320,60 @@ private void _validateObjectDefinitionSettings(
 		}
 	}
 
+	private void _validateObjectDefinitionWorkflowLinks(
+			long companyId, String scope,
+			List<WorkflowDefinitionLink> workflowDefinitionLinks,
+			List<ObjectDefinitionSetting> objectDefinitionSettings)
+		throws PortalException {
+
+		if (!FeatureFlagManagerUtil.isEnabled(companyId, "LPD-17564") ||
+			workflowDefinitionLinks.isEmpty() ||
+			Objects.equals(scope, ObjectDefinitionConstants.SCOPE_COMPANY)) {
+
+			return;
+		}
+
+		for (WorkflowDefinitionLink workflowDefinitionLink :
+				workflowDefinitionLinks) {
+
+			if (workflowDefinitionLink.getGroupId() == 0) {
+				continue;
+			}
+
+			ObjectScopeProvider objectScopeProvider =
+				_objectScopeProviderRegistry.getObjectScopeProvider(scope);
+
+			if (!objectScopeProvider.isValidGroupId(
+					workflowDefinitionLink.getGroupId())) {
+
+				throw new ObjectDefinitionScopeException(
+					"An object definition can only be link to a workflow " +
+						"definition within the same scope");
```

Comment:
"An object definition can only be **linked** to a workflow definition within the same scope."